contracts/ColonyFunding.sol

```diff
@@ -141,6 +142,86 @@ contract ColonyFunding is ColonyStorage, DSMath {
     return nonRewardPotsTotal[_token];
   }
 
+  function getGlobalRewardPayoutCount() public view returns (uint256) {
+    return globalRewardPayoutCount;
+  }
+
+  function getUserRewardPayoutCount(address _user) public view returns (uint256) {
+    return userRewardPayoutCount[_user];
+  }
+
+  function startNextRewardPayout(address _token) public auth returns(uint256) {
+    globalRewardPayoutCount += 1;
+
+    //TODO: Lock everyones tokens
+
+    uint256 amount = pots[0].balance[_token];
+    pots[0].balance[_token] = 0;
+
+    rewardPayoutCycles[globalRewardPayoutCount] = RewardPayoutCycle(
+      IColonyNetwork(colonyNetworkAddress).getReputationRootHash(),
+      amount,
+      amount,
+      _token,
+      block.number
+    );
+
+    return globalRewardPayoutCount;
+  }
+
+  function claimRewardPayout(uint8 _payoutId, int256 _userReputation, int256 _totalReputation) public {
+    require(block.number - rewardPayoutCycles[_payoutId].blockNumber < 500000);
+    require(_payoutId - userRewardPayoutCount[msg.sender] == 1);
+
+    //TODO: Prove that userReputation and totalReputation in reputationState are correct
+
+    address tokenAddress = rewardPayoutCycles[_payoutId].token;
+    ERC20Extended payoutToken = ERC20Extended(tokenAddress);
+    uint256 userTokens = payoutToken.balanceOf(msg.sender);
+    uint256 totalTokens = sub(payoutToken.totalSupply(), payoutToken.balanceOf(address(this)));
```

Comment:
1. The tokens used to work out the reward a user is entitled to should be the colony tokens they hold in the colony paying out the reward, not the tokens of the type being paid out that they already hold.

2. I hadn't considered removing the colony tokens held by the colony from consideration from this calculation. I think that this will give strictly better behaviour than what was spec'd (everyone should get slightly more rewards than they would have done before - i.e. there will be less left over and returned to the pot). Very nice!